painting/src/main/java/com/ataulm/artcollector/painting/ui/PaintingActivity.kt

```diff
@@ -27,12 +34,46 @@ class PaintingActivity : AppCompatActivity() {
         val paintingId = intent.data!!.pathSegments.last()
         injectDependencies(PaintingId(paintingId))
 
+        // TODO: check with app links that it still works and doesn't care about postponed transitions
+        postponeEnterTransition()
+
+        intent.loadImageIfAvailable()
         viewModel.painting.observe(this, DataObserver<Painting> { painting ->
             title = painting.title
             titleArtistTextView.text = "${painting.title} // ${painting.artist.name}" // TODO: string format, but also PaintingUiModel
             creditLineTextView.text = painting.creditLine ?: "Harvard Art Museum"
             creditLineTextView.setOnClickListener { startActivity(webIntent(painting.webUrl)) }
-            picasso.load(painting.imageUrl).into(imageView)
+            painting.loadImageIfDifferent()
         })
     }
+
+    private fun Intent.loadImageIfAvailable() {
+        imageUrl()?.let {
+            glideRequestManager
+                    .load(it)
+                    .listener(startTransitionRequestListener)
+                    .into(imageView)
+        }
+    }
+
+    private fun Painting.loadImageIfDifferent() {
+        if (imageUrl != null && intent.imageUrl() != imageUrl) {
+            glideRequestManager
+                    .load(imageUrl)
+                    .listener(startTransitionRequestListener)
+                    .into(imageView)
+        }
+    }
+
+    private val startTransitionRequestListener = object : RequestListener<Drawable> {
+        override fun onLoadFailed(e: GlideException?, model: Any?, target: Target<Drawable>?, isFirstResource: Boolean): Boolean {
+            startPostponedEnterTransition()
```

Comment:
regardless whether the image load fails or is successful, we need to resume the postponed transition! else our activity will just hang there, lifeless, still.